### ISSUE TITLE
Some command line usage errors don't print messages

### ISSUE THREAD

**Issue by baldurk** [KhronosGroup/glslang#3719]
Some mistakes in invoking glslang on the command line aren't clearly reported, only re-printing the whole usage text string with no explanation of what went wrong.

The one I ran into is I was trying to compile an hlsl shader and e.g. `glslang -H test.hlsl` doesn't work as it doesn't specify the shader stage. It's an obvious error in retrospect but I had to debug glslang and breakpoint on `usage()` to see why.

Some other errors are reported, e.g. `glslang -gVS test.frag` will print:

```
glslang.exe: Error: cannot generate debug information unless linking to generate code (use -h for usage)
```

In general I think any call to `usage()` from the command line tool as a result of an error should print what the error is. Looking at the source it seems a few of the resource-shifting and backing-type change options would also fail opaquely in this way.

**Comment by arcady-lunarg** [comment]
Yeah, this is a fair point and these usability issues are something that needs to be addressed. I'll look into it, it should be pretty easy to fix these.
